zimply/zim_core.py

```diff
@@ -1000,6 +1006,21 @@ def suggest(self, query, start=0, end=9, separator=" "):
     def get_suggestions_results_count(self, query, separator=" "):
         return self.get_search_results_count(query, separator)
 
+    def _tokenize_search_query(self, query, separator=" "):
+        tokens = []
+        quote = False
+
+        while query:
+            token, _part, query = query.partition('"')
+            token = token.strip()
+            if token and quote:
+                tokens.append(token.strip())
+            elif token:
+                tokens.extend(token.strip().split(separator))
+            quote = quote is False and _part == '"'
```

Comment:
This while might have less performance than a regular expression but it's doing the job.

Reply:
Thanks for checking. Performance is probably not something we should care _too_ much about. Most ZIM files have a Xapian index nowadays, and the biggest impact on the user would be when performing a request, which is only a very short string. Creating the FTS index of course takes time, but since this happens (or can happen) in the background I am not too concerned about the performance and prefer readability over an undecipherable regular expression.

Reply:
> Most ZIM files have a Xapian index nowadays

Actually, they even have two: one for the article titles and one fulltext

Reply:
Very true 😄 . The version2 branch of ZIMply defaults to the fulltext search, then falls back to the title search. If neither is available (not included or no Xapian support) then the fallback sqlite index is built and used.